common/src/main/java/io/pravega/common/ExceptionHelpers.java

```diff
@@ -60,8 +60,8 @@ public static boolean shouldUnwrap(Class<? extends Exception> c) {
      * @return The cause or the exception provided.
      */
     public static Throwable unwrapIfRequired(Throwable e) {
-        if ((e instanceof CompletionException || e instanceof ExecutionException) && e.getCause() != null) {
-            return e.getCause();
+        while ((e instanceof CompletionException || e instanceof ExecutionException) && e.getCause() != null) {
```

Comment:
Orthogonal comment: this method seems similar to getRealException with only difference being IO exception. 
do we need two methods?